pkg/varlinkapi/images.go

```diff
@@ -48,10 +57,188 @@ func (i *LibpodAPI) ListImages(call ioprojectatomicpodman.VarlinkCall) error {
 	return call.ReplyListImages(imageList)
 }
 
+// GetImage returns a single image in the form of a ImageInList
+func (i *LibpodAPI) GetImage(call ioprojectatomicpodman.VarlinkCall, name string) error {
+	runtime, err := libpodruntime.GetRuntime(i.Cli)
+	if err != nil {
+		return call.ReplyRuntimeError(err.Error())
+	}
+	newImage, err := runtime.ImageRuntime().NewFromLocal(name)
+	if err != nil {
+		return call.ReplyErrorOccurred(err.Error())
+	}
+	labels, _ := newImage.Labels(getContext())
+	containers, _ := newImage.Containers()
+	size, _ := newImage.Size(getContext())
+
+	il := ioprojectatomicpodman.ImageInList{
+		Id:          newImage.ID(),
+		ParentId:    newImage.Parent,
+		RepoTags:    newImage.Names(),
+		RepoDigests: newImage.RepoDigests(),
+		Created:     newImage.Created().String(),
+		Size:        int64(*size),
+		VirtualSize: newImage.VirtualSize,
+		Containers:  int64(len(containers)),
+		Labels:      labels,
+	}
+	return call.ReplyGetImage(il)
+}
+
 // BuildImage ...
-// TODO Waiting for buildah to be vendored into libpod to do this only one
-func (i *LibpodAPI) BuildImage(call ioprojectatomicpodman.VarlinkCall) error {
-	return call.ReplyMethodNotImplemented("BuildImage")
+func (i *LibpodAPI) BuildImage(call ioprojectatomicpodman.VarlinkCall, config ioprojectatomicpodman.BuildInfo) error {
+	var (
+		memoryLimit int64
+		memorySwap  int64
+	)
+
+	runtime, err := libpodruntime.GetRuntime(i.Cli)
+	if err != nil {
+		return call.ReplyRuntimeError(err.Error())
+	}
+	defer runtime.Shutdown(false)
+
+	systemContext := types.SystemContext{}
+	dockerfiles := config.Dockerfile
+	contextDir := ""
+
+	for i := range dockerfiles {
+		if strings.HasPrefix(dockerfiles[i], "http://") ||
+			strings.HasPrefix(dockerfiles[i], "https://") ||
+			strings.HasPrefix(dockerfiles[i], "git://") ||
+			strings.HasPrefix(dockerfiles[i], "github.com/") {
```

Comment:
Ew... Can we just force people to prepend `https://`, `http://`, `git://` if it's remote, and always assume local otherwise?

Reply:
this isnt me ... this is buildah. sorry dude

Reply:
could it be a regex to avoid the multiple checks?